Use open dialog when cloning on windows

tidy-dev opened the pull request:
Closes #12812

## Description
Reverted clone dialog to from save to open dialog on windows.

### Screenshots
https://user-images.githubusercontent.com/75402236/159717300-078251d4-0e98-4b44-8e96-c580dc29fd11.mov

## Release notes
Notes: [Fixed] Clone dialog "choose" button uses an open dialog for directory selection on mac.

tidy-dev committed:
Use open dialog on windows

billygriffin reviewed (commented):
Awesome! Just a question - do we use the feature flag pattern for differentiating between Windows and macOS anywhere else? I'm wondering if this will be clear for others who come upon this in the future.

tidy-dev committed:
Use Darwin flag with documentation

tidy-dev commented:
> Awesome! Just a question - do we use the feature flag pattern for differentiating between Windows and macOS anywhere else? I'm wondering if this will be clear for others who come upon this in the future.

Not really - we do have one other feature flag with arm64 updating that is flagged to macOs.  This save vs open dialog is more of unique case because we are providing different feature per OS, where usually macOS vs windows is either wording or because some underlying process (like spawn) doesn't work the same.

That said, I moved it into the code so that it would be more visible to a future devs that we are doing something different per OS and added some documentation as to why since it is a unique case.

Other notes: I took out the feature flag since it isn't really providing anything at this point and we could easily add back since we are still flagging by os if we so choose.

j-f1 commented on app/src/lib/feature-flag.ts:
```diff
@@ -80,7 +80,7 @@ export function enableUpdateFromEmulatedX64ToARM64(): boolean {
 
 /** Should we allow using the save dialog when choosing where to clone a repo */
 export function enableSaveDialogOnCloneRepository(): boolean {
```
Is this function used anymore? If not, should it be removed?

tidy-dev replied:
Nope! Good catch. I thought I did remove it, but didn't save that file. 😳

j-f1 reviewed (dismissed):
This makes a lot of sense — especially after seeing the new dialog on Windows, it much better matches the current macOS experience!

tidy-dev committed:
Remove unused feature flag

tidy-dev committed:
... remove all remnants of feature flag ...

Nonso15 commented:
Legit company to invest and trade your Bitcoin and earn your profits daily safely withdrawal instantly.
🎲🎲🎲🎲🎲🎲🎲🎲🎲
https://Jomo-trade.com
https://Jomo-trade.com

Legit✅✅
No withdrawal fee✅✅
Instant withdraw ✅✅
https://Jomo-trade.com
https://Jomo-trade.com

sergiou87 reviewed (approved):
🐸 🕴️ ❗
